storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -454,76 +621,61 @@ func (s *fileSystemSource) loadShardReadersDataIntoShardResult(
 		}
 	}
 
-	s.handleErrorsAndUnfulfilled(
-		resultLock, bootstrapResult, shard, tr, shardResult, timesWithErrors)
+	s.handleErrorsAndUnfulfilled(resultLock, runResult, shard,
+		tr, shardResult, timesWithErrors)
 }
 
-func (s *fileSystemSource) ReadData(
+func (s *fileSystemSource) read(
 	md namespace.Metadata,
 	shardsTimeRanges result.ShardTimeRanges,
-	_ bootstrap.RunOptions,
-) (result.DataBootstrapResult, error) {
-	if shardsTimeRanges.IsEmpty() {
-		return result.NewDataBootstrapResult(), nil
-	}
-
+	run runType,
+) (*runResult, error) {
 	var (
 		nsID              = md.ID()
-		blockRetrieverMgr = s.opts.DatabaseBlockRetrieverManager()
+		seriesCachePolicy = s.opts.ResultOptions().SeriesCachePolicy()
 		blockRetriever    block.DatabaseBlockRetriever
 	)
-	if blockRetrieverMgr != nil {
-		s.log.WithFields(
-			xlog.NewField("namespace", nsID.String()),
-		).Infof("filesystem bootstrapper resolving block retriever")
-
-		var err error
-		blockRetriever, err = blockRetrieverMgr.Retriever(md)
-		if err != nil {
-			return nil, err
-		}
-
-		s.log.WithFields(
-			xlog.NewField("namespace", nsID.String()),
-			xlog.NewField("shards", len(shardsTimeRanges)),
-		).Infof("filesystem bootstrapper caching block retriever shard indices")
-
-		shards := make([]uint32, 0, len(shardsTimeRanges))
-		for shard := range shardsTimeRanges {
-			shards = append(shards, shard)
-		}
-
-		err = blockRetriever.CacheShardIndices(shards)
-		if err != nil {
-			return nil, err
-		}
+	if shardsTimeRanges.IsEmpty() {
+		return newRunResult(), nil
 	}
 
-	switch s.opts.ResultOptions().SeriesCachePolicy() {
-	case series.CacheAll:
-		// No checks necessary
-	case series.CacheAllMetadata:
-		// Need to check block retriever available
-		if blockRetriever == nil {
-			return nil, fmt.Errorf(
-				"missing block retriever when using series cache metadata for namespace: %s",
-				nsID.String())
+	if run == bootstrapDataRunType {
```

Comment:
do you want to retain the logging statements from the original? either way is fine, just calling out the difference

Reply:
actually nvm, see that you're doing this in `resolveBlockRetrieverAndCacheDataShardIndices`